addons/website/static/src/js/components/survey/survey.js

```diff
@@ -0,0 +1,807 @@
+            }
+            if (themes[1]) {
+              state.themeSelection.secondTheme.name = themes[1].name;
+              state.themeSelection.secondTheme.url = themes[1].url;
+              state.themeSelection.secondTheme.id = themes[1].id;
+            }
+            if (themes[2]) {
+              state.themeSelection.thirdTheme.name = themes[2].name;
+              state.themeSelection.thirdTheme.url = themes[2].url;
+              state.themeSelection.thirdTheme.id = themes[2].id;
+            }
+          }
+        };
+
+        const getters = {
+          getWebsiteTypes({state}) {
+            return Object.values(state.description.descriptionData.websiteTypes).map(x => x);
+          },
+
+          getSelectedType({state}, id) {
+            return id ? state.description.descriptionData.websiteTypes[id] : undefined;
+          },
+
+          getWebsitePurpose({state}) {
+            return Object.values(state.description.descriptionData.websitePurposes).map(x => x);
+          },
+
+          getSelectedPurpose({state}, id) {
+            return id ? state.description.descriptionData.websitePurposes[id] : undefined;
+          },
+
+          getFeatures({state}) {
+            const columnNumber = 3;
+            const featureRows = [];
+
+            const features = Object.values(state.featureSelection.features);
+            let currentRow = [];
+            for (let i = 0; i < features.length; i += 1) {
+              currentRow.push(features[i]);
+              if (currentRow.length === columnNumber) {
+                featureRows.push(currentRow);
+                currentRow = [];
+              }
+            }
+            if (currentRow.length > 0) {
+              const rowLength = currentRow.length;
+              for (let i = 0; i < (columnNumber - rowLength); i += 1) {
+                currentRow.push({
+                  type: 'empty'
+                });
+              }
+              featureRows.push(currentRow);
+            }
+            return featureRows;
+          },
+
+          getPalettes({state}) {
+            const columnNumber = 4;
+            const paletteRows = [];
+            let palettes = Object.values(state.paletteSelection.palettes);
+            let currentRow = [];
+            for (let i = 0; i < palettes.length; i += 1) {
+              currentRow.push(palettes[i]);
+              if (currentRow.length === columnNumber) {
+                paletteRows.push(currentRow);
+                currentRow = [];
+              }
+            }
+            if (currentRow.length > 0) {
+              const rowLength = currentRow.length;
+              for (let i = 0; i < (columnNumber - rowLength); i += 1) {
+                currentRow.push({
+                  type: 'empty'
+                });
+              }
+              paletteRows.push(currentRow);
+            }
+            return paletteRows;
+          },
+        };
+
+        async function getInitialState(wid) {
```

Comment:
Any reason this function is async? Seems you're not using await anywhere (although as said earlier, maybe you should use await instead of then when possible for consistency)